src/Components/MediaPresenter.ts

```diff
@@ -68,55 +56,41 @@ export class MediaPresenter {
   }
 
   @mobx.action
-  setDuration(index: number, duration: number): void {
-    this.durations[index] = duration;
+  setDuration(fileIndex: number, duration: number): void {
+    this.durations[fileIndex % this.filesLength] = duration;
```

Comment:
why do you use`%` (remainder) here?

Reply:
(and all other uses)

Reply:
This is to ensure that it goes around the file list. (like a loop, repeating the files :D )

Reply:
Is this still to do with the Modal?

Reply:
this is to do how the modal shows the media. I have removed it for now because it is less related to the homepage :D